lib/logs.js

```diff
@@ -0,0 +1,249 @@
+/*
+Copyright 2016 Balena
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as querystring from 'querystring';
+
+import { EventEmitter } from 'events';
+import * as ndjson from 'ndjson';
+import { AbortController as AbortControllerPonyfill } from 'abortcontroller-polyfill/dist/cjs-ponyfill';
+import { findCallback } from './util';
+import { globalEnv } from './util/global-env';
+
+const AbortController =
+	globalEnv != null && 'AbortController' in globalEnv
+		? globalEnv.AbortController
+		: AbortControllerPonyfill;
```

Comment:
This changed slightly to make typescript happy but the effect should be equivalent, previous was
```coffee
AbortController = globalEnv?.AbortController || AbortControllerPonyfill